eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -620,31 +314,6 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: A
     nodeParams.db.network.getNode(remoteNodeId).flatMap(_.addresses.headOption.map(_.socketAddress))
   }
 
-  /**
-   * Peer may want to filter announcements based on timestamp.
-   *
-   * @param gossipTimestampFilter_opt optional gossip timestamp range
-   * @return
-   *         - true if this is our own gossip
-   *         - true if there is a filter and msg has no timestamp, or has one that matches the filter
-   *         - false otherwise
-   */
-  def timestampInRange(msg: RoutingMessage, origins: Set[GossipOrigin], gossipTimestampFilter_opt: Option[GossipTimestampFilter]): Boolean = {
-    // For our own gossip, we should ignore the peer's timestamp filter.
-    val isOurGossip = msg match {
-      case n: NodeAnnouncement if n.nodeId == nodeParams.nodeId => true
-      case _ if origins.contains(LocalGossip) => true
-      case _ => false
-    }
-    // Otherwise we check if this message has a timestamp that matches the timestamp filter.
-    val matchesFilter = (msg, gossipTimestampFilter_opt) match {
-      case (_, None) => false // BOLT 7: A node which wants any gossip messages would have to send this, otherwise [...] no gossip messages would be received.
-      case (hasTs: HasTimestamp, Some(GossipTimestampFilter(_, firstTimestamp, timestampRange))) => hasTs.timestamp >= firstTimestamp && hasTs.timestamp <= firstTimestamp + timestampRange
-      case _ => true // if there is a filter and message doesn't have a timestamp (e.g. channel_announcement), then we send it
-    }
-    isOurGossip || matchesFilter
-  }
-
   // a failing channel won't be restarted, it should handle its states
```

Comment:
nit: the comment isn't completely accurate anymore, we also have some `Client` children (and we should also let them stop themselves without restarting them).

Reply:
It wasn't accurate before, we were already creating `Client` children.